hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleWorkflow.java

```diff
@@ -337,31 +348,17 @@ private Stream<SingleTransactionRecord> execute(@NonNull final UserTxn userTxn)
      *
      * @return the failure record
      */
-    private Stream<SingleTransactionRecord> failInvalidRecordStream(@NonNull final UserTxn userTxn) {
-        final var failInvalidRecordListBuilder = new RecordListBuilder(userTxn.consensusNow());
-        final var recordBuilder = failInvalidRecordListBuilder.userTransactionRecordBuilder();
-        initializeUserRecord(recordBuilder, userTxn.txnInfo());
-        recordBuilder.status(FAIL_INVALID);
+    private Stream<SingleTransactionRecord> failInvalidStreamItems(@NonNull final UserTxn userTxn) {
         userTxn.stack().rollbackFullStack();
-        return recordStream(userTxn, failInvalidRecordListBuilder);
-    }
-
-    /**
-     * Has the side effect of adding a {@link ResponseCodeEnum#BUSY} record to
-     * the transaction's record stream, but skips any other execution.
-     *
-     * @param userTxn the user transaction to skip
-     */
-    private void skip(@NonNull final UserTxn userTxn) {
-        final TransactionInfo txnInfo = userTxn.txnInfo();
-        userTxn.recordListBuilder()
-                .userTransactionRecordBuilder()
-                .transaction(txnInfo.transaction())
-                .transactionBytes(txnInfo.signedBytes())
-                .transactionID(txnInfo.txBody().transactionIDOrElse(TransactionID.DEFAULT))
-                .exchangeRate(exchangeRateManager.exchangeRates())
-                .memo(txnInfo.txBody().memo())
-                .status(BUSY);
+        final var failInvalidBuilder = new SingleTransactionRecordBuilderImpl(REVERSIBLE, NOOP_RECORD_CUSTOMIZER, USER);
+        initializeBuilderInfo(failInvalidBuilder, userTxn.txnInfo())
+                .status(FAIL_INVALID)
+                .consensusTimestamp(userTxn.consensusNow());
+        recordCache.add(
+                userTxn.creatorInfo().nodeId(),
+                requireNonNull(userTxn.txnInfo().payerID()),
+                List.of(failInvalidBuilder.build()));
```

Comment:
_Nit_ - could `build()` this just once.

Reply:
Ah yes .. fixed. Thanks @tinker-michaelj